pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -88,15 +88,20 @@ func (w *watcher) Run(ctx context.Context) error {
 	defer os.RemoveAll(workingDir)
 	w.workingDir = workingDir
 
-	for _, repoCfg := range w.config.Repositories {
-		repo, err := w.cloneRepo(ctx, repoCfg)
+	repoCfgs := w.config.GetRepositoryMap()
+	for _, r := range w.config.EventWatcher.GitRepos {
+		cfg, ok := repoCfgs[r.RepoID]
+		if !ok {
+			return fmt.Errorf("repo id %q doesn't exist in the Piped repositories list", r.RepoID)
+		}
+		repo, err := w.cloneRepo(ctx, cfg)
 		if err != nil {
 			return err
 		}
 		defer repo.Clean()
 
 		w.wg.Add(1)
-		go w.run(ctx, repo, repoCfg)
```

Comment:
nit, Add log to mention which repository and how many repositories are loaded could be good for future debugging. Can place it here or in the `run` function body.

Reply:
For sure. I just added something like it.